File: orchestrator-java-reactive/src/main/java/com/github/mzr/coder/orchestrator/java/reactive/service/impl/RiskService1Impl.java
package com.github.mzr.coder.orchestrator.java.reactive.service.impl;

import com.github.mzr.coder.model.ServiceRequest;
import com.github.mzr.coder.model.ServiceResponse;
import com.github.mzr.coder.orchestrator.java.reactive.config.UrlConfig;
import com.github.mzr.coder.orchestrator.java.reactive.service.RiskService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

@Slf4j
@RequiredArgsConstructor
@Service
public class RiskService1Impl implements RiskService {

    private final WebClient.Builder webClientBuilder;
    private final UrlConfig urlConfig;

    @Override
    public String getName() {
        return "service1";
    }

    @Override
    public Mono<ServiceResponse> execute(ServiceRequest request) {
        var name = request.getName();
        var mono = Mono.just(request);
        log.info("Trying get info from {} for request with name {}", getName(), name);
        return webClientBuilder.baseUrl(urlConfig.getService1())
                .build()
                .post()
                .uri("/info")
                .body(mono, ServiceRequest.class)
                .retrieve()
                .bodyToMono(ServiceResponse.class)
                .doOnSuccess(sr ->
                        log.info("Got info from {} for request with name {} and approve {}",
                                getName(),
                                name,
                                sr.getApprove()
                        )
                );
    }
}

File: orchestrator-kotlin-coroutine/src/main/java/com/github/mzr/coder/orchestrator/kotlin/reactive/service/impl/RiskService1Impl.kt
package com.github.mzr.coder.orchestrator.kotlin.reactive.service.impl

import com.github.mzr.coder.model.ServiceRequest
import com.github.mzr.coder.model.ServiceResponse
import com.github.mzr.coder.orchestrator.kotlin.reactive.config.UrlConfig
import com.github.mzr.coder.orchestrator.kotlin.reactive.service.RiskService
import kotlinx.coroutines.reactor.awaitSingle
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

@Service
class RiskService1Impl(
    private val webClientBuilder: WebClient.Builder,
    private val urlConfig: UrlConfig
) : RiskService {

    private val log = LoggerFactory.getLogger(RiskService1Impl::class.java)

    override val name: String = "service1"

    override suspend fun execute(request: ServiceRequest): ServiceResponse {
        val name = request.name
        log.info("Trying get info from {} for request with name {}", this.name, name)
        val serviceResponse = webClientBuilder!!.baseUrl(urlConfig.service1)
            .build()
            .post()
            .uri("/info")
            .bodyValue(request)
            .retrieve()
            .bodyToMono(ServiceResponse::class.java)
            .awaitSingle()
        log.info("Got info from {} for request with name {} and approve {}", this.name, name, serviceResponse.approve)
        return serviceResponse

    }
}